### ISSUE TITLE
`aten.unsqueeze.default` fallback with input_shape [1, 64] and output_shape [1, 64, 1]

### ISSUE THREAD

**Issue by swimdi** [tenstorrent/pytorch2.0_ttnn#386]
In llama model, [there has an input varation](https://github.com/tenstorrent/pytorch2.0_ttnn/blob/196ae752825cab6c532fa8ed741f3acd030385a3/tests/autogen_op/Llama/test_Llama_aten_unsqueeze_default.py#L41C9-L41C53)
```
["Tensor<[1, 64]> self = ?", "int dim = 2"]
```
Which will reshape the tensor from `[1, 64]` to `[1, 64, 1]`

Currently this input variation will not convert to `ttnn.reshape`, and if force convert to `ttnn.reshape`,  will raise runtime error
```
Failed to run. Raised exception: TT_FATAL @ /tmp/build-via-sdist-n2_feo7u/metal_libs-0.53.0rc28+wormhole.b0/ttnn/cpp/ttnn/operations/data_movement/data_transfer/data_transfer.cpp:23: input_tensor.get_legacy_shape()[-1] * input_tensor.element_size() % sizeof(uint32_t) == 0
```

Is there any ttnn op can handle this case?

**Comment by swimdi** [comment]
The fall back situation of unsqueeze is [from there](https://github.com/tenstorrent/pytorch2.0_ttnn/blob/6d0167785d6fcbd72f1e463eb967d30b695e558f/torch_ttnn/passes/lowering/to_tt_pass.py#L636)
```
            if node.target == torch.ops.aten.unsqueeze.default:
                output_size = node.meta["val"].size()
                output_size = list(output_size)
                if output_size[-1] == args[0].meta["val"].size()[-1]:
                    return g.call_function(ttnn.reshape, args=(args[0], output_size))
                return None
```
And the following input variation is all the fall back case
```
        ["Tensor<[16]> self = ?", "int dim = 1"],
        ["Tensor<[23]> self = ?", "int dim = -1"],
        ["Tensor<[1, 23, 40]> self = ?", "int dim = 3"],
        ["Tensor<[15]> self = ?", "int dim = 1"],
        ["Tensor<[2]> self = ?", "int dim = 1"],
        ["Tensor<[17]> self = ?", "int dim = 1"],
        ["Tensor<[30]> self = ?", "int dim = 1"],
        ["Tensor<[60]> self = ?", "int dim = 1"],
        ["Tensor<[120]> self = ?", "int dim = 1"],
        ["Tensor<[240]> self = ?", "int dim = 1"],
        ["Tensor<[480]> self = ?", "int dim = 1"],
        ["Tensor<[1, 64]> self = ?", "int dim = 2"],
        ["Tensor<[1, 512]> self = ?", "int dim = 2"],
        ["Tensor<[3]> self = ?", "int dim = 1"],
        ["Tensor<[320]> self = ?", "int dim = 1"],
        ["Tensor<[3234]> self = ?", "int dim = 1"],
        ["Tensor<[1, 320]> self = ?", "int dim = 2"],
        ["Tensor<[1, 640]> self = ?", "int dim = 2"],
        ["Tensor<[1, 1280]> self = ?", "int dim = 2"],
        ["Tensor<[16]> self = ?", "int dim = -1"],
        ["Tensor<[32]> self = ?", "int dim = -1"],
        ["Tensor<[64]> self = ?", "int dim = -1"],
        ["Tensor<[12]> self = ?", "int dim = -1"],
        ["Tensor<[1, 5, 1, 16]> self = ?", "int dim = 4"],
        ["Tensor<[1, 1, 1, 16]> self = ?", "int dim = 4"],
        ["Tensor<[56]> self = ?", "int dim = -1"],
        ["Tensor<[28]> self = ?", "int dim = -1"],
        ["Tensor<[14]> self = ?", "int dim = -1"],
        ["Tensor<[7]> self = ?", "int dim = -1"],
        ["Tensor<[800]> self = ?", "int dim = 1"],
        ["Tensor<[50]> self = ?", "int dim = -1"],
        ["Tensor<[100]> self = ?", "int dim = -1"],
        ["Tensor<[0]> self = ?", "int dim = 1"],
        ["Tensor<[24]> self = ?", "int dim = 1"],
        ["Tensor<[300]> self = ?", "int dim = 1"],
        ["Tensor<[8732]> self = ?", "int dim = 1"],
        ["Tensor<[25]> self = ?", "int dim = 1"],
        ["Tensor<[64, 49]> self = ?", "int dim = 2"],
        ["Tensor<[16, 49]> self = ?", "int dim = 2"],
        ["Tensor<[4, 49]> self = ?", "int dim = 2"],
        ["Tensor<[64, 64]> self = ?", "int dim = 2"],
        ["Tensor<[16, 64]> self = ?", "int dim = 2"],
        ["Tensor<[4, 64]> self = ?", "int dim = 2"],
        ["Tensor<[10]> self = ?", "int dim = 1"],
```

You can run `pytest tests/autogen_op/ALL/test_ALL_aten_unsqueeze_default.py --input_var_check_ttnn` to reproduce it

**Comment by jvegaTT** [comment]
The following reshapes in TILE format all have the equivalent behaviour to a transpose (dimensions are unpadded tensor dimensions and a,b,c,x are any number):
[x] or [1,x] or [1,1,x] or [1,1,1,x] -> [a,1] or [a,b,1] or [a,b,c,1]
or
[a,1] or [a,b,1] or [a,b,c,1] -> [x] or [1,x] or [1,1,x] or [1,1,1,x]
Note all reshapes must produce the same sized tensor so a*b*c=x must be imposed

The following cases from above match the format that allows a transpose to be done:
        ["Tensor<[16]> self = ?", "int dim = 1"],
        ["Tensor<[23]> self = ?", "int dim = -1"],
        ["Tensor<[15]> self = ?", "int dim = 1"],
        ["Tensor<[2]> self = ?", "int dim = 1"],
        ["Tensor<[17]> self = ?", "int dim = 1"],
        ["Tensor<[30]> self = ?", "int dim = 1"],
        ["Tensor<[60]> self = ?", "int dim = 1"],
        ["Tensor<[120]> self = ?", "int dim = 1"],
        ["Tensor<[240]> self = ?", "int dim = 1"],
        ["Tensor<[480]> self = ?", "int dim = 1"],
        ["Tensor<[1, 64]> self = ?", "int dim = 2"],
        ["Tensor<[1, 512]> self = ?", "int dim = 2"],
        ["Tensor<[3]> self = ?", "int dim = 1"],
        ["Tensor<[320]> self = ?", "int dim = 1"],
        ["Tensor<[3234]> self = ?", "int dim = 1"],
        ["Tensor<[1, 320]> self = ?", "int dim = 2"],
        ["Tensor<[1, 640]> self = ?", "int dim = 2"],
        ["Tensor<[1, 1280]> self = ?", "int dim = 2"],
        ["Tensor<[16]> self = ?", "int dim = -1"],
        ["Tensor<[32]> self = ?", "int dim = -1"],
        ["Tensor<[64]> self = ?", "int dim = -1"],
        ["Tensor<[12]> self = ?", "int dim = -1"],
        ["Tensor<[1, 1, 1, 16]> self = ?", "int dim = 4"],
        ["Tensor<[56]> self = ?", "int dim = -1"],
        ["Tensor<[28]> self = ?", "int dim = -1"],
        ["Tensor<[14]> self = ?", "int dim = -1"],
        ["Tensor<[7]> self = ?", "int dim = -1"],
        ["Tensor<[800]> self = ?", "int dim = 1"],
        ["Tensor<[50]> self = ?", "int dim = -1"],
        ["Tensor<[100]> self = ?", "int dim = -1"],
        ["Tensor<[24]> self = ?", "int dim = 1"],
        ["Tensor<[300]> self = ?", "int dim = 1"],
        ["Tensor<[8732]> self = ?", "int dim = 1"],
        ["Tensor<[25]> self = ?", "int dim = 1"],
        ["Tensor<[10]> self = ?", "int dim = 1"],
This tensor is a special case that should simply return what was supplied
        ["Tensor<[0]> self = ?", "int dim = 1"],
The following tensors require a more complex fix to get working:
        ["Tensor<[64, 49]> self = ?", "int dim = 2"],
        ["Tensor<[16, 49]> self = ?", "int dim = 2"],
        ["Tensor<[4, 49]> self = ?", "int dim = 2"],
        ["Tensor<[64, 64]> self = ?", "int dim = 2"],
        ["Tensor<[16, 64]> self = ?", "int dim = 2"],
        ["Tensor<[4, 64]> self = ?", "int dim = 2"],
        ["Tensor<[1, 5, 1, 16]> self = ?", "int dim = 4"],
        ["Tensor<[1, 23, 40]> self = ?", "int dim = 3"],

We need a function to check for the above cases. The call to this function should be done in the function 
ttnn::Tensor ReshapeViewOperation::invoke(const ttnn::Tensor& tensor, const ttnn::Shape& shape)
in the file
ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape.cpp
right after the lines that say:
```
// First Case, No reshape Required
    if (tensor_shape == shape) {
        return tensor;
    }
```

Refer to the transpose OP to perform a transpose in the kernel if the case matches. Transpose should be simple as it is a 2D column to a 2D row or a 2D row to a 2D column which means the elements in tile 0 are still in tile 0 but transposed, tile 1 is still in tile 1 but transposed and so on. You should NOT be calling ttnn::transpose as that is more general but has slow host transfers. The kernel API transpose_wh_tile in: 
tt_metal/include/compute_kernel_api/transpose_wh.h
will do a transpose to a single tile. You just need to add a wrapper to do this operation for all the tiles in the system and copy the result to the same tile that provided the data and wrap all of that in a simple kernel code that is scheduled to run for the above cases.

**Comment by swimdi** [comment]
Fallback issue solved in #450
